IPs/RSKIPxxx.md

```diff
@@ -0,0 +1,262 @@
+The Powpeg ordinals address derivation starts with the creation of a custom redeem script that pushes the verOwner field. The verOwner field is defines as:
+
+* **version**: specifies the derivation version (1 byte) 
+* **owner**: ordinal owner address (a 20 byte RSK address)
+
+Therefore verOwner consist of a 21-byte binary chunk. The verOwner field is pushed and immediately dropped from the stack using the `OP_DROP` opcode, so the pushed data does not alter the processing of the redeem script.
+
+The redeem script for Powpeg-derived addresses is a modification of the standard Powpeg redeem script. We show how to build such scripts, but we omit the emergency multisig script path for clarity. The whole script must be considered when computing the address. The new custom redeem script has the following format:
+
+```scriptPubKey: OP_HASH160 <redeemScriptHash> OP_EQUAL
+scriptSig: OP_0 <signatures> <redeemScript>
+redeemScript: <verOwner> OP_DROP OP_M <publicKeys> OP_N OP_CHECKMULTISIG
+```
+
+## Main Ordinals Address Derivation
+
+The mechanism is the same as the previous one, but setting the version field to 0, and without setting the address, resulting in a push of the single byte 0.
+
+`redeemScript: PUSH_0 OP_DROP OP_M <publicKeys> OP_N OP_CHECKMULTISIG`
+
+## Gas costs
+
+The folllwing list provide the gas costs of the new methods:
+
+* **getFederationOrdinalsAddress**(): 500
+* **registerBtcTransactionWithOrdinals**(): 22000
+* **registerBtcTransactionWithOrdinalsDMFA**(): 25000
+* **getOrdinalOwner**(): 800 
+* **transferOrdinal**(): 6000
+* **releaseOrdinal**(): 23000
+* **getOrdinalsAnnualParkingFee**(): 200
+* **payParkingExtensionFee**(): 6000
+* **getOrdinalParkingExtensionFee**(): 800
+* **getOrdinalParkingExpirationDate**(): 800
+
+# Rationale
+
+## Security 
+
+We believe that while some ordinals and inscriptions could be highly valued, due to the low cost of creating spammy inscriptions, in the long term most ordinals will be low valued. The cost of an ordinal may become much lower than the BTC amount of an average bitcoin peg-in, and even lower than the cost of a UTXO consolidation. This has two consequences: first, ordinals should pay a rent to avoid spam, and second, ordinals can be carefully isolated with their own security model. 
+While the double-spend of an ordinal during peg-in can cause distress to its owner, it does not affect any other user of Roostock, not any other ONFT. There can’t be a run to the peg if an ordinal is lost, while a mismatch of BTC collateral in Bitcoin and rBTC  issued in Roostock can be the cause of such an event. 
+
+Since ordinals can be tracked independently in separate UTXOs, the double-spend of an ordinal does not affect the peg funds, nor the other ordinals. Therefore, we can lower the number of confirmations required for an ordinal peg-in. We propose that ordinals require 6 Bitcoin confirmations for peg-in (about 1 hour), and 120 confirmations (about 1 hour) for peg-out. 
```

Comment:
This will imply a change in POWHSM as well. It cannot discern between regular peg-outs and ordinal peg-outs.
Should it?

Reply:
Do you have an idea of how easy/hard is to allow this? Can the HSM distinguish between an ordinal peg-out and a btc peg-out ? It could use the output value to distinguish, yet there is the danger that it accepts a 120-conf peg-out that drains all the funds in miner fees, so it must also use the amount of the fees to distinguish, which makes the process much more complicated, because the HSM doesn't know the amounts of the inputs. Input amounts are only available explicitly in segwit transactions.  

A better and more secure option is that the Powpeg uses a different set of public keys to manage ordinals, so one set of keys manage 120-conf peg.outs, and another set, 4000-conf peg.outs.